Tests/ApolloCacheDependentTests/WatchQueryTests.swift

```diff
@@ -149,11 +149,101 @@ class WatchQueryTests: XCTestCase {
       expectation = self.expectation(description: "Updated after fetching other query")
 
       client.fetch(query: HeroNameQuery(), cachePolicy: .fetchIgnoringCacheData)
-      
+
       waitForExpectations(timeout: 5, handler: nil)
     }
   }
-  
+
+  func testWatchedQueryGetsUpdatedWithListReorderingFromOtherQuery() throws {
+    let initialRecords: RecordSet = [
+      "QUERY_ROOT": ["hero": Reference(key: "QUERY_ROOT.hero")],
+      "QUERY_ROOT.hero": [
+        "name": "R2-D2",
+        "__typename": "Droid",
+        "friends": [
+          Reference(key: "QUERY_ROOT.hero.friends.0"),
+          Reference(key: "QUERY_ROOT.hero.friends.1"),
+          Reference(key: "QUERY_ROOT.hero.friends.2")
+        ]
+      ],
+      "QUERY_ROOT.hero.friends.0": ["__typename": "Human", "name": "Luke Skywalker"],
+      "QUERY_ROOT.hero.friends.1": ["__typename": "Human", "name": "Han Solo"],
+      "QUERY_ROOT.hero.friends.2": ["__typename": "Human", "name": "Leia Organa"],
+      ]
+
+    withCache(initialRecords: initialRecords) { (cache) in
+      let networkTransport = MockNetworkTransport(body: [
+        "data": [
+          "hero": [
+            "name": "Artoo",
+            "__typename": "Droid",
+            "friends": [
+              ["__typename": "Human", "name": "Luke Skywalker"],
+              ["__typename": "Human", "name": "Leia Organa"],
+            ]
+          ]
+        ]
+        ])
+      let store = ApolloStore(cache: cache)
+      let client = ApolloClient(networkTransport: networkTransport, store: store)
+
+      let query = HeroAndFriendsNamesQuery()
+
+      var verifyResult: GraphQLResultHandler<HeroAndFriendsNamesQuery.Data>
+
+      verifyResult = { result in
+        switch result {
+        case .success(let graphQLResult):
+          XCTAssertNil(graphQLResult.errors)
+          guard let data = graphQLResult.data else {
+            XCTFail("No data in graphQLResult!")
+            return
+          }
+
+          XCTAssertEqual(data.hero?.name, "R2-D2")
+          let friendsNames = data.hero?.friends?.compactMap { $0?.name }
+          XCTAssertEqual(friendsNames, ["Luke Skywalker", "Han Solo", "Leia Organa"])
+        case .failure(let error):
+          XCTFail("Unexpected error: \(error)")
+        }
+      }
+
+      var expectation = self.expectation(description: "Fetching query")
+
+      _ = client.watch(query: query) { result in
+        verifyResult(result)
+        expectation.fulfill()
+      }
+
+      waitForExpectations(timeout: 5, handler: nil)
+
+      verifyResult = { result in
```

Comment:
I think the ordering here is throwing me wayyyy off - so the first `verifyResult` is verifying what's already in the initial records, and this second one is verifying the result after the the request you set up with `networkTransport` actually hits the cache? 

I realize due to the construction of `withCache` this may not be possible to reorder, but some comments in here would probably make this a LOT easier to follow.

Reply:
Thanks for the feedback, I added a few comments.